crates/interpreter/src/instructions/macros.rs

```diff
@@ -205,6 +211,12 @@ macro_rules! as_usize_or_fail {
             $interp.instruction_result = $reason;
             return;
         }
-        x[0] as usize
+        match ::core::convert::TryInto::<usize>::try_into(x[0]) {
+            Ok(val) => val,
+            Err(_) => {
+                $interp.instruction_result = $reason;
+                return;
+            }
+        }
```

Comment:
```suggestion
        let Ok(val) = match ::core::convert::TryInto::<usize>::try_into(x[0])  else {
                $interp.instruction_result = $reason;
                return;
        }
        val
```

We can use `let else` here

Reply:
`let else` needs a newer version of rust (>=1.65), that means we need to give up the old users. Does it make sense?
https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html